packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -0,0 +1,142 @@
+import { SmartAccountClient } from "permissionless";
+import { EntryPoint } from "permissionless/_types/types";
+import { SmartAccount } from "permissionless/accounts";
+import {
+    Abi,
+    Address,
+    BaseError,
+    Chain,
+    ContractFunctionRevertedError,
+    PublicClient,
+    TransactionReceipt,
+    Transport,
+} from "viem";
+
+import { CrossmintSDKError, WalletErrorCode } from "@crossmint/client-sdk-base";
+
+export type TransactionServiceTransactionRequest = {
+    address: Address;
+    abi: Abi;
+    functionName: string;
+    args: any;
+    value?: bigint;
+};
+
+/**
+ * Error thrown when a transaction fails to send.
+ * @param viemError The error thrown by the viem client.
+ */
```

Comment:
Would it be clear from these errors whether the error originated during simulation, sending or confirming? If not, it may make sense to add that as a parameter in the error

Reply:
I added two subclasses, `SendTransactionSimulationError` and `SendTransactionConfirmError` for errors that are thrown in those respective stages